.gitlab-ci.yml

```diff
@@ -15,137 +16,189 @@ variables:
+  - substrate-contracts-node-rand-extension --tmp --dev -lruntime::contracts=debug > /tmp/substrate-contracts-node-rand-extension.log 2>&1 &
+
+.shutdown-substrate-contracts-node:                  &shutdown-substrate-contracts-node
+  - pkill -f "substrate-contracts-node --tmp --dev"
 
-.start-canvas-rand-extension:      &start-canvas-rand-extension
-  - canvas-rand-extension --tmp --dev -lruntime::contracts=debug > /tmp/canvas-rand-extension.log 2>&1 &
+.shutdown-substrate-contracts-node-rand-extension:   &shutdown-substrate-contracts-node-rand-extension
+  - pkill -f "substrate-contracts-node-rand-extension --tmp --dev"
 
-.build-contracts:                  &build-contracts
+
+# needed vars have to be "exported" on a earlier stage
+parent-vars:
+  stage:                                             declare-vars
+  <<:                                                *kubernetes-env
+  script:
+    - BRANCH="master"
+    - REPO="https://github.com/paritytech/ink.git"
+    - REPO_NAME="ink"
+    - echo ${TRGR_REF}
+    - if [ -n "$TRGR_REF" ] && [ "$TRGR_REF" != "master" ]; then
+        PR_JSON=`curl -s https://api.github.com/repos/paritytech/ink/pulls/${TRGR_REF}`;
+        BRANCH=`echo "${PR_JSON}" | jq -r .head.ref`;
+        REPO=`echo "${PR_JSON}" | jq -r .head.repo.git_url`;
+        REPO_NAME=`echo "${PR_JSON}" | jq -r .head.repo.name`;
+      fi
+    - echo "BRANCH=${BRANCH}" | tee -a parent-vars.env
+    - echo "REPO=${REPO}" | tee -a parent-vars.env
+    - echo "REPO_NAME=${REPO_NAME}" | tee -a parent-vars.env
+    - echo "CSV_NAME=${REPO_NAME}_${BRANCH}" | tee -a parent-vars.env
+    - echo "REDIS_RESULTS_KEY=${CI_PROJECT_NAME}::results::{$REPO_NAME}::${BRANCH}" | tee -a parent-vars.env
+    - echo "REDIS_RESULTS_KEY_MASTER=${CI_PROJECT_NAME}::results::{$REPO_NAME}::master" | tee -a parent-vars.env
+  artifacts:
+    reports:
+      dotenv: parent-vars.env
+
+
+.build-contracts:                                    &build-contracts
+  # delete old list items if the key has existed previously
+  - redis-cli -u $GITLAB_REDIS_URI del $REDIS_RESULTS_KEY
+  - echo 'Data will be written to $REDIS_RESULTS_KEY'
   - for example in ${INK_EXAMPLES_PATH}/*/; do
-      echo "cargo contract build --manifest-path=${example}/Cargo.toml" >> /tmp/cmds;
+      echo "./.scripts/build-contract.sh ${example} | redis-cli -u ${GITLAB_REDIS_URI} -x rpush ${REDIS_RESULTS_KEY}" >> /tmp/cmds;
     done
   - for contract in ${DELEGATOR_SUBCONTRACTS}; do
-      echo "cargo contract build --manifest-path=${INK_EXAMPLES_PATH}/delegator/${contract}/Cargo.toml" >> /tmp/cmds;
+      echo "./.scripts/build-contract.sh ${INK_EXAMPLES_PATH}/delegator/${contract} | redis-cli -u ${GITLAB_REDIS_URI} -x rpush ${REDIS_RESULTS_KEY}" >> /tmp/cmds;
     done
   - parallel -j 2 < /tmp/cmds
+  # all ci/cd keys need to have ttl
+  - redis-cli -u $GITLAB_REDIS_URI expire $REDIS_RESULTS_KEY $GITLAB_REDIS_TTL
```

Comment:
Can you elaborate on this? Also, what's the TTL value we'd be working with?

Reply:
`2419200`, 28 days.

Redis is in-memory storage, so we need some way to GC keys. Non-TTL keys have a chance to be evicted under memory pressure anyway. Explicitly specifying TTL here tells people that their keys can't be "eternal" in Redis.

I'm going to GC all non-TTL keys in the Redis on a scheduled basis to keep the Redis "tidy". 

Do you need more TTL or any other eviction strategy here?

Reply:
Gotcha, thanks for the clarification

> Do you need more TTL or any other eviction strategy here?

No, this seems fine. We plan to use these keys pretty quickly anyways (by posting a comment in the PR), so there's no need for these to stick around for long periods of time.